frontend/src/components/BCDataGrid/components/Editors/AsyncValidationEditor.jsx

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect, forwardRef, useImperativeHandle } from 'react'
+import { useDebounce } from '@/utils/debounce'
+
+export const AsyncValidationEditor = forwardRef(
+  ({ value, onValueChange, eventKey, rowIndex, column, ...props }, ref) => {
+    const [inputValue, setInputValue] = useState(value)
+    const [valid, setValid] = useState(true)
+    const [validating, setValidating] = useState(false)
+    const [touched, setTouched] = useState(false)
+
+    const debouncedInputVal = useDebounce(inputValue, props.debounceLimit)
+
+    function inputHandler(e) {
+      setTouched(true)
+      setInputValue(e.target.value)
+      onValueChange(e.target.value)
+      setValidating(true)
+    }
+
+    useEffect(() => {
+      // random time between 0 and 1000ms
+      const timeout = Math.floor(Math.random() * props.debounceLimit)
```

Comment:
whats the use case for a random value instead of a fixed value like 1000?

Reply:
modified.